ossl/src/digest.rs

```diff
@@ -37,6 +37,17 @@ impl EvpMd {
     pub unsafe fn as_mut_ptr(&mut self) -> *mut EVP_MD {
         self.ptr
     }
+
+    /// Tries to clone the `EvpMd`.
+    pub fn try_clone(&self) -> Result<EvpMd, Error> {
+        let success = unsafe { EVP_MD_up_ref(self.ptr) == 1 };
```

Comment:
pleae use:
```
let ret = ..
if ret != 1 {
   return Err..
}
Ok(..)
```

We manage all the errors this way and it's easier to read code when it is consistent